ceremonies/src/lib.rs

```diff
@@ -82,18 +78,49 @@ decl_storage! {
+
+        ReputableRegistry get(fn reputable_registry): double_map hasher(blake2_128_concat) CommunityCeremony, hasher(blake2_128_concat) ParticipantIndexType => T::AccountId;
+        ReputableIndex get(fn reputable_index): double_map hasher(blake2_128_concat) CommunityCeremony, hasher(blake2_128_concat) T::AccountId => ParticipantIndexType;
+        ReputableCount get(fn reputable_count): map hasher(blake2_128_concat) CommunityCeremony => ParticipantIndexType;
+
+        EndorseeRegistry get(fn endorsee_registry): double_map hasher(blake2_128_concat) CommunityCeremony, hasher(blake2_128_concat) ParticipantIndexType => T::AccountId;
+        EndorseeIndex get(fn endorsee_index): double_map hasher(blake2_128_concat) CommunityCeremony, hasher(blake2_128_concat) T::AccountId => ParticipantIndexType;
+        EndorseeCount get(fn endorsee_count): map hasher(blake2_128_concat) CommunityCeremony => ParticipantIndexType;
+
+        NewbieRegistry get(fn newbie_registry): double_map hasher(blake2_128_concat) CommunityCeremony, hasher(blake2_128_concat) ParticipantIndexType => T::AccountId;
+        NewbieIndex get(fn newbie_index): double_map hasher(blake2_128_concat) CommunityCeremony, hasher(blake2_128_concat) T::AccountId => ParticipantIndexType;
+        NewbieCount get(fn newbie_count): map hasher(blake2_128_concat) CommunityCeremony => ParticipantIndexType;
+
+
+        AllowedBootstrapperCount get(fn allowed_bootstrapper_count): map hasher(blake2_128_concat) CommunityCeremony => ParticipantIndexType;
+        AllowedReputableCount get(fn allowed_reputable_count): map hasher(blake2_128_concat) CommunityCeremony => ParticipantIndexType;
+        AllowedEndorseeCount get(fn allowed_endorsee_count): map hasher(blake2_128_concat) CommunityCeremony => ParticipantIndexType;
+        AllowedNewbieCount get(fn allowed_newbie_count): map hasher(blake2_128_concat) CommunityCeremony => ParticipantIndexType;
+
+
+        S1BootstrappersReputables get(fn s1_bootstrappers_reputables): map hasher(blake2_128_concat) CommunityCeremony => u64;
+        S1Endorsees get(fn s1_endorsees): map hasher(blake2_128_concat) CommunityCeremony => u64;
+        S1Newbies get(fn s1_newbies): map hasher(blake2_128_concat) CommunityCeremony => u64;
+
+        S2BootstrappersReputables get(fn s2_bootstrappers_reputables): map hasher(blake2_128_concat) CommunityCeremony => u64;
+        S2Endorsees get(fn s2_endorsees): map hasher(blake2_128_concat) CommunityCeremony => u64;
+        S2Newbies get(fn s2_newbies): map hasher(blake2_128_concat) CommunityCeremony => u64;
+
+
+        MBootstrappersReputables get(fn m_bootstrappers_reputables): map hasher(blake2_128_concat) CommunityCeremony => u64;
+        MEndorsees get(fn m_endorsees): map hasher(blake2_128_concat) CommunityCeremony => u64;
+        MNewbies get(fn m_newbies): map hasher(blake2_128_concat) CommunityCeremony => u64;
```

Comment:
Please introduce a new struct in primitives/ceremonies
```
pub struct AssignmentParams {
  pub m: u64,
  pub s1: u64,
  pub s2: u64
}
```
and derive all usual traits, see other examples

And simplify this storage definition and all function signatures accordingly